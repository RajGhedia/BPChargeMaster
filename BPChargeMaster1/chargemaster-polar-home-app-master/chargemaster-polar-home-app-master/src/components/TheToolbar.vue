<template>
    <v-toolbar>
        <v-container pa-0 ma-0>
            <v-layout>
                <v-flex xs2 pt-4 text-xs-center>
                    <v-icon v-if="showMenu" @click.stop="openDrawer"
                        >menu</v-icon
                    >
                    <router-link v-if="showBack" :to="backTo">
                        <v-icon>chevron_left</v-icon>
                    </router-link>
                </v-flex>
                <v-flex xs8 text-xs-center class="title" py-4>
                    <slot></slot>
                </v-flex>
                <v-flex xs2 pt-4 text-xs-center>
                    <router-link v-if="showClose" :to="backTo">
                        <v-icon>close</v-icon>
                    </router-link>
                </v-flex>
            </v-layout>
        </v-container>
    </v-toolbar>
</template>
<script>
import EventBus from '@/event-bus'
export default {
    props: {
        showMenu: Boolean,
        showClose: Boolean,
        showBack: Boolean,
        backTo: String
    },
    methods: {
        openDrawer() {
            EventBus.$emit('openDrawer')
        }
    }
}
</script>
<style>
.v-image__image--cover {
    background-size: initial !important;
}

.title {
    font-size: 1.3rem;
}

a {
    text-decoration: none;
}
</style>
