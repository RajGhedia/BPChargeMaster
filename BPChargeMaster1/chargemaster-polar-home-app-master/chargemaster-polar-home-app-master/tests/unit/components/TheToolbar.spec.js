import TheToolbar from '@/components/TheToolbar.vue'
import Vue from 'vue'
import Vuetify from 'vuetify'
import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'

const localVue = createLocalVue()
Vue.use(Vuetify)

describe('TheToolbar.vue', () => {
    it('renders correctly', () => {
        const wrapper = mount(TheToolbar, {
            localVue,
            propsData: {
                showMenu: true
            },
            stubs: { 'router-link': RouterLinkStub },
            slots: {
                default: 'A page title would go here'
            }
        })

        expect(wrapper.element).toMatchSnapshot()

        wrapper.destroy()
    })
})
