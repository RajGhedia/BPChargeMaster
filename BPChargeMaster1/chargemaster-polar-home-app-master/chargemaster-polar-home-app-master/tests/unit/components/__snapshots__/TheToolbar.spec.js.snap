// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheToolbar.vue renders correctly 1`] = `
<nav
  class="v-toolbar theme--light"
  style="margin-top: 0px; padding-right: 0px; padding-left: 0px; transform: translateY(0px);"
>
  <div
    class="v-toolbar__content"
    style="height: 64px;"
  >
    <div
      class="container pa-0 ma-0"
    >
      <div
        class="layout"
      >
        <div
          class="flex xs2 pt-4 text-xs-center"
        >
          <i
            aria-hidden="true"
            class="v-icon v-icon--link material-icons theme--light"
          >
            menu
          </i>
           
          <!---->
        </div>
         
        <div
          class="flex title xs8 text-xs-center py-4"
        >
          A page title would go here
        </div>
         
        <div
          class="flex xs2 pt-4 text-xs-center"
        >
          <!---->
        </div>
      </div>
    </div>
  </div>
</nav>
`;
